pkg/resources/deployments.go

```diff
@@ -0,0 +1,178 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package resources
+
+import (
+	operatorv1alpha1 "github.com/ibm/ibm-licensing-hub-operator/pkg/apis/operator/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var replicas = int32(1)
+
+var log = logf.Log.WithName("controller_licenseadvisor_deployment")
+
+func GetLicensingDeployment(instance *operatorv1alpha1.IBMLicensingHub) *appsv1.Deployment {
+	metaLabels := LabelsForLicensingHubMeta(instance)
+	selectorLabels := LabelsForLicensingHubSelector(instance)
+	podLabels := LabelsForLicensingHubPod(instance)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetResourceName(instance),
+			Namespace: instance.GetNamespace(),
+			Labels:    metaLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selectorLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      podLabels,
+					Annotations: AnnotationsForPod(),
+				},
+				Spec: corev1.PodSpec{
+					Volumes: getLicensingHubVolumes(instance.Spec),
+					Containers: []corev1.Container{
+						GetDatabaseContainer(instance.Spec),
+						GetReceiverContainer(instance.Spec),
+					},
+					TerminationGracePeriodSeconds: &seconds60,
+					ServiceAccountName:            GetServiceAccountName(instance),
+					Affinity: &corev1.Affinity{
+						NodeAffinity: &corev1.NodeAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+								NodeSelectorTerms: []corev1.NodeSelectorTerm{
+									{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      "beta.kubernetes.io/arch",
+												Operator: corev1.NodeSelectorOpIn,
+												Values:   []string{"amd64"},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					Tolerations: []corev1.Toleration{
+						{
+							Key:      "dedicated",
+							Operator: corev1.TolerationOpExists,
+							Effect:   corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:      "CriticalAddonsOnly",
+							Operator: corev1.TolerationOpExists,
+						},
+					},
+				},
+			},
+		},
+	}
+	return deployment
+}
+
+func ShouldUpdateDeployment(expectedDeployment *appsv1.Deployment, foundDeployment *appsv1.Deployment) bool {
```

Comment:
I think we do not use this code, some looking for upadates is already in controller

Reply:
ok, it will be fixed